File: backend/schemas/broker.py
from pydantic import BaseModel


class BrokerConfigCreate(BaseModel):
    broker_name: str
    api_key: str
    api_secret: str
    redirect_url: str


class BrokerConfigResponse(BaseModel):
    broker_name: str
    api_key_masked: str
    api_secret_masked: str
    redirect_url: str
    is_active: bool


class BrokerListItem(BaseModel):
    name: str
    display_name: str
    supported_exchanges: list[str]
    is_configured: bool = False
    is_active: bool = False

File: frontend/src/types/broker.ts
export interface BrokerListItem {
  name: string;
  display_name: string;
  supported_exchanges: string[];
  is_configured: boolean;
  is_active: boolean;
}

export interface BrokerConfigData {
  broker: string;
  api_key: string;
  api_secret: string;
  redirect_url: string;
}

export interface BrokerConfigResponse {
  status: string;
  message: string;
}

export interface BrokerRedirectResponse {
  url: string;
}
